### PR TITLE
Reorganizing the Rubicon_ml docs

### PR DESCRIPTION
**Pull request by shania-m**
closes: #224 #190 

---

## What
  * Reorganized the Rubicon documentation into `How to guides`, `tutorials`, `explanation`, and `reference`.
  * Added the outstanding `faq` issues into the `explanation`
  * Added example to tag failed experiments

## How to Test
  * All notebook examples run
  * Docs build successfully

### CODE REVIEW

**Review by ryanSoley — COMMENTED** [review]
I left some comments on the FAQ pertaining to issue #209. however, before we get any further, I think we need to reorganize some things here. imo this PR has gotten to the point where there are too many different things going on.

I have comments on both the failed experiment example (addressing #190) as well as the docs organization (#208) that need to be addressed but if I put them all on this one PR its going to become overwhelming.

here's how I propose we move forward - and please feel free to disagree and propose another plan if you want:
1. address the FAQ comments I've left, then cut a new branch from main and copy only the new faq.rst file over to it. then open a new PR from that branch that only addresses #209 so we can get that merged in and out of the way
2. cut a new branch from main and copy only the new notebook example file over to it. then open a new PR from that branch that only addresses #190. that way I can explain my comments without them getting in the way of the FAQ/organizational work
3. once those are both addressed and merged, rebase this branch off of those changes and have this PR only be organizational stuff

File: docs/source/faqs.rst [hunk 1]
```diff
@@ -106,3 +106,27 @@ unit of work that you'd like to compare multiple runs of. In fact, we've used
...
+
+Like MLFlow, rubicon-ml also aims to offer integrations with commonly used tools, such as `Scikit-learn <https://capitalone.github.io/rubicon-ml/integrations/integration-sklearn.html>`_. Again, rubicon-ml tries to be less prescriptive in these integrations. For example, `MLFlow says exactly what it will log <https://mlflow.org/docs/latest/tracking.html#scikit-learn-experimental>`_  when using with Scikit-learn. rubicon-ml has a set of defaults that are logged to each estimator, but also `supports user-defined loggers <https://capitalone.github.io/rubicon-ml/library_reference.html#rubicon_ml.sklearn.RubiconPipeline>`_ for any estimator, like the built-in `FilterEstimatorLogger <https://github.com/capitalone/rubicon-ml/blob/main/rubicon_ml/sklearn/filter_estimator_logger.py>`_.
+
+If you've got any more specific questions feel free to ask! It's also been a bit since I've done a full, hands-on analysis of MLFlow's capabilities, so if you've used it more recently and anything I've said seems incorrect, please let me know!
```
[the remark this answers is not in the document]
Comment (andreafehrman) [reply docs/source/faqs.rst:124]: ^

**Review by andreafehrman — DISMISSED** [review]
Looks great!

**Comment by ryanSoley** [comment]
@shania-m @andreafehrman sorry about letting this PR get so stale. I've collected some thoughts around where to go with this and added them to the original issue
> https://github.com/capitalone/rubicon-ml/issues/208

feel free to comment on that, otherwise we'll sync up tomorrow to discuss

**Pull request closed without merging**